src/figuresClasses.ts

```diff
@@ -2,10 +1,98 @@
+interface Figure {
+  shape: string;
+  color: string;
 
-export class Triangle implements Figure {}
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: string;
+
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Sides length must be greater than 0');
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('Sides length cannot form a triangle');
+    }
+
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+    this.shape = 'triangle';
+  }
 
-export class Circle implements Figure {}
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
 
-export class Rectangle implements Figure {}
+    return Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+  }
+}
+
+export class Circle implements Figure {
+  shape: string;
+
+  constructor(
+    public color: string,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Radius must be greater than 0');
```

Comment:
When throwing an error, provide a more detailed description. For example, 'Radius must be greater than 0' could be expanded to explain why a positive radius is required.